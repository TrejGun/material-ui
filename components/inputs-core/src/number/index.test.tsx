import { FC, ReactNode } from "react";
import { IntlProvider } from "react-intl";
import { MemoryRouter } from "react-router-dom";
import { cleanup, render } from "@testing-library/react";
import { FormProvider, useForm } from "react-hook-form";
import { createTheme, ThemeProvider } from "@mui/material";
import fetch from "node-fetch";

import { TestIdProvider } from "@gemunion/provider-test-id";

import { NumberInput } from "./index";

beforeAll(() => (window.fetch = window.fetch || fetch));
afterEach(cleanup);

const i18n = {
  "form.labels.number": "Number",
  "form.placeholders.number": "50",
};

const AllTheProviders: FC<{ children: ReactNode }> = ({ children }) => {
  return (
    <MemoryRouter>
      <ThemeProvider theme={createTheme()}>
        <IntlProvider locale="en" messages={i18n}>
          <TestIdProvider testId="number">
            <FormProvider {...useForm()}>{children}</FormProvider>
          </TestIdProvider>
        </IntlProvider>
      </ThemeProvider>
    </MemoryRouter>
  );
};

describe("<NumberInput />", () => {
  it("renders positive value", () => {
    const props = {
      name: "number",
      value: 50,
    };

    const { asFragment } = render(<NumberInput {...props} />, { wrapper: AllTheProviders });

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders negative value", () => {
    const props = {
      name: "number",
      value: -50,
    };

    const { asFragment } = render(<NumberInput {...props} />, { wrapper: AllTheProviders });

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders decimal value", () => {
    const props = {
      name: "number",
      value: 9.99,
    };

    const { asFragment } = render(<NumberInput {...props} />, { wrapper: AllTheProviders });

    expect(asFragment()).toMatchSnapshot();
  });
});
