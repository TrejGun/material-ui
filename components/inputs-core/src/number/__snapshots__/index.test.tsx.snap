// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NumberInput /> renders decimal value 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-1nin81k-MuiFormControl-root-MuiTextField-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
      for=":r2:"
      id=":r2:-label"
    >
      Number
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1a1fmpi-MuiInputBase-root-MuiInput-root"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
        data-testid="number-number"
        id=":r2:"
        name="number"
        placeholder="50"
        type="number"
        value="9.99"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`<NumberInput /> renders negative value 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-1nin81k-MuiFormControl-root-MuiTextField-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
      for=":r1:"
      id=":r1:-label"
    >
      Number
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1a1fmpi-MuiInputBase-root-MuiInput-root"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
        data-testid="number-number"
        id=":r1:"
        name="number"
        placeholder="50"
        type="number"
        value="-50"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`<NumberInput /> renders positive value 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-1nin81k-MuiFormControl-root-MuiTextField-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
      for=":r0:"
      id=":r0:-label"
    >
      Number
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1a1fmpi-MuiInputBase-root-MuiInput-root"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
        data-testid="number-number"
        id=":r0:"
        name="number"
        placeholder="50"
        type="number"
        value="50"
      />
    </div>
  </div>
</DocumentFragment>
`;
